events/actions.go

```diff
@@ -0,0 +1,42 @@
+package events
+
+import (
+	"encoding/json"
+)
+
+type Action interface {
+	Type() string
+	SignerAddress() string
+}
+
+// Exports this action to a map
+func ActionToMap(a any) (map[string]any, error) {
```

Comment:
The implementations here are kinda funny, but the more I think about it they're kind of right... I mean, there's a technically-unnecessary JSON serialization, but we do want to populate the structs based on the JSON tags os this is _almost_ correct behavior.